dataflow/gemma-flex-template/Dockerfile

```diff
@@ -0,0 +1,50 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This uses Ubuntu with Python 3.10 and comes with CUDA drivers for
+# GPU use.
+ARG SERVING_BUILD_IMAGE=pytorch/pytorch:2.1.2-cuda11.8-cudnn8-runtime
+
+FROM ${SERVING_BUILD_IMAGE}
+
+WORKDIR /workspace
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends apt-utils \
+    && apt-get install -y --no-install-recommends curl \
+    && apt-get install -y --no-install-recommends wget \
+    && apt-get install -y --no-install-recommends git \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY requirements.txt requirements.txt
+RUN pip install --upgrade pip \
+    && pip install --no-cache-dir -r requirements.txt 
+
+RUN git clone https://github.com/google/gemma_pytorch.git
+RUN pip install ./gemma_pytorch
```

Comment:
This should be `pip install --no-cache-dir ./gemma_pytorch` to reduce image size.

Also I noticed that this installs `torch-2.3.1` while this container is using `pytorch/pytorch:2.1.2-cuda11.8-cudnn8-runtime`, so I'm not sure if there are any incompatibilities. This might require pinning some versions or some other way to make sure the versions match.

It might be a good idea to have some sort of assert somewhere to make sure the versions match.